apps/web/src/components/notebook/index.js

```diff
@@ -154,6 +155,18 @@ const menuItems = [
     iconColor: "error",
     icon: Icon.Trash,
     onClick: async ({ items }) => {
+      if (await showDeleteNotesDialog("Notebook")) {
+        items.map(async (item) => {
+          db.notebooks.notebook(item.id).topics.all.map(async (topic) => {
+            await db.notebooks
+              .notebook(item.id)
+              .topics.topic(topic.id)
+              .all.map(async (note) => {
+                await db.notes.delete(note.id);
```

Comment:
Instead of deleting the note directly, it'll be better to collect all the notes in an array and then use `Multiselect.moveNotesToTrash`. This takes care of all the edge cases.